### PR TITLE
LB-387: MessyBrainz: Add API endpoints to fetch MSIDs or MBIDs.

### PR DESCRIPTION
**Pull request by kartikeyaSh**
# Summary
Add API endpoints to get all MBIDs and MSIDs using a given MSID
and another endpoint to get all the MSIDs using a given MBID.

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
We have created clusters now we need API endpoints to make use of the clustered data.

* JIRA ticket (_optional_): [LB-387](https://tickets.metabrainz.org/browse/LB-387)

# Solution

Added two API endpoints one to access information using MSIDs. And another one to access information using MBIDs.

### CODE REVIEW

File: messybrainz/db/recording.py [hunk 1]
```diff
@@ -148,3 +148,48 @@ def create_recording_clusters():
...
+        "msid": msid
+    })
+
+    if cluster_id.rowcount:
+        return cluster_id.fetchone()['cluster_id']
+    else:
+        return None
+
+
+def get_gids_using_cluster_id(connection, cluster_id):
```
Comment (mayhem) [inline messybrainz/db/recording.py:170]: Change to get_recording_msids_from_cluster_id ?

File: messybrainz/db/recording.py [hunk 2]
```diff
@@ -148,3 +148,48 @@ def create_recording_clusters():
...
+          FROM recording_cluster
+         WHERE cluster_id = :cluster_id
+    """), {
+        "cluster_id": cluster_id
+    })
+
+    return [r[0] for r in result]
+
+
+def get_mbids_using_cluster_id(connection, cluster_id):
```
Comment (mayhem) [inline messybrainz/db/recording.py:184]: s/using/from/ ?

File: messybrainz/db/release.py [hunk 3]
```diff
@@ -396,3 +396,17 @@ def fetch_and_store_releases_for_all_recording_mbids():
                 pass
 
         return num_recording_mbids_processed, num_recording_mbids_added
+
+
+def get_msids_using_cluster_id(connection, cluster_id):
```
Comment (mayhem) [inline messybrainz/db/release.py:401]: get_release_msids_from_cluster_id ?

File: messybrainz/webserver/views/api.py [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
-
+from uuid import UUID
 import messybrainz
 import messybrainz.db.exceptions
 import ujson
 
+REQUEST_TYPE = ['artist', 'recording', 'release']
+INCLUDES = ['mbid']
+
 api_bp = Blueprint('api', __name__)
```
Comment (mayhem) [inline messybrainz/webserver/views/api.py:18]: We should use /ws/1 as the prefix for the API.

File: messybrainz/webserver/views/api.py [hunk 5]
```diff
@@ -55,3 +63,111 @@ def get_aka(messybrainz_id):
...
+                    fetched_mbids = db_release.get_release_mbids_using_msid(connection, cluster_id)
+
+        data = {'msid': fetched_msids}
+        if mbid:
+            data['mbid'] = fetched_mbids
+    except BadDataException:
+        raise BadRequest
+    except NoDataFoundException:
+        raise NotFound
+
```
Comment (mayhem) [inline messybrainz/webserver/views/api.py:111]: There are likely other exceptions the DB is going to throw -- we should catch more exceptions here.

File: messybrainz/webserver/views/api.py [hunk 6]
```diff
@@ -55,3 +63,111 @@ def get_aka(messybrainz_id):
...
+@api_bp.route("/msid", methods=['GET'])
+@crossdomain()
+def get_msid_using_msid():
+    """Returns all MessyBrainz IDs for the MSID in request."""
+
+    msid = request.args.get('id')
+    request_type = request.args.get('request_type')
+    mbid = False
+    includes = request.args.getlist('includes')
```
Comment (mayhem) [inline messybrainz/webserver/views/api.py:76]: At the summit we agreed to no longer use includes, but resource selectors. Take a look at this document:

https://docs.google.com/document/d/1ClnGKq3rsagERpoAjCjf3Jl1DoCfolzlgj0OXmAy9Pg/edit?usp=sharing

But, I don't see includes being used -- for now just remove them please.

**Review by mayhem — CHANGES_REQUESTED** [review]
I've left some comments that need to be addressed, but we really need to discuss the API structure for MSB before we continue working on this PR.

**Comment by paramsingh** [comment]
Closing this as the repository has been merged into [listenbrainz-server](https://github.com/metabrainz/listenbrainz-server).

**Pull request closed without merging**